app/models/metasploit/credential/pkcs12.rb

```diff
@@ -0,0 +1,84 @@
+require 'openssl'
+require 'base64'
+
+# A private Pkcs12 file.
+class Metasploit::Credential::Pkcs12 < Metasploit::Credential::Private
+  #
+  # Attributes
+  #
+
+  # @!attribute data
+  #   A private pkcs12 file, base64 encoded - i.e. starting with 'MIIMhgIBAzCCDFAGCSqGSIb3DQEHAaCC....'
+  #
+  #   @return [String]
+
+  #
+  #
+  # Validations
+  #
+  #
+
+  #
+  # Attribute Validations
+  #
+
+  validates :data,
+            presence: true
+  #
+  # Method Validations
+  #
+
+  validate :readable
+
+  #
+  # Instance Methods
+  #
+
+  # The {#data key data}'s fingerprint, suitable for displaying to the
+  # user.
+  #
+  # @return [String]
+  def to_s
+    return '' unless data
+
+    cert = openssl_pkcs12.certificate
+    result = []
+    result << "subject:#{cert.subject.to_s}"
+    result << "issuer:#{cert.issuer.to_s}"
+    result.join(',')
```

Comment:
Any way we could expand on this to also shown when a PKCS12 entry is going to expire? OpenSSL should be able to show this information already and it would be useful to have this in here as well.

Reply:
I believe this could be added with `result << "expires:#{cert.not_after}"`, but I wonder if it's useful to show to the user by default? I'm imagining that the average expiry for stored certs would be far enough in the future that the expiry metadata would just be clutter in the UI

Of course let me know if that's a bad assumption 😄

Reply:
They usually expire after a year by default. Whether or not we want to display this is debatable but could be useful particularly since such data might help in the cleanup after an engagement.

Reply:
Will punt on this to reduce the visual noise for now :+1: